recipes/make/all/conanfile.py

```diff
@@ -1,64 +1,82 @@
+        return getattr(self, "settings_build", self.settings)
+
+    def export_sources(self):
+        export_conandata_patches(self)
 
     def configure(self):
-        del self.settings.compiler.libcxx
-        del self.settings.compiler.cppstd
+        self.settings.rm_safe("compiler.cppstd")
+        self.settings.rm_safe("compiler.libcxx")
 
-    def source(self):
-        tools.get(**self.conan_data["sources"][self.version],
-                  destination=self._source_subfolder, strip_root=True)
+    def layout(self):
+        basic_layout(self, src_folder="src")
 
     def package_id(self):
         del self.info.settings.compiler
 
-    def build(self):
-        for patch in self.conan_data.get("patches").get(self.version, []):
-            tools.patch(**patch)
+    def source(self):
+        get(self, **self.conan_data["sources"][self.version],
+            destination=self.source_folder, strip_root=True)
+
+    def generate(self):
+        if is_msvc(self):
+            vcvars = VCVars(self)
+            vcvars.generate()
+        if self._settings_build.os != "Windows":
+            tc = AutotoolsToolchain(self)
+            tc.generate()
 
-        with tools.chdir(self._source_subfolder):
+    def build(self):
+        apply_conandata_patches(self)
+        with chdir(self, self.source_folder):
             # README.W32
-            if tools.os_info.is_windows:
-                if self.settings.compiler == "Visual Studio":
+            if self._settings_build.os == "Windows":
+                if is_msvc(self):
                     command = "build_w32.bat --without-guile"
                 else:
                     command = "build_w32.bat --without-guile gcc"
             else:
-                env_build = AutoToolsBuildEnvironment(self)
-                env_build.configure()
+                autotools = Autotools(self)
+                autotools.configure()
                 command = "./build.sh"
-            with tools.vcvars(self.settings) if self.settings.compiler == "Visual Studio" else tools.no_op():
-                self.run(command)
+            self.run(command)
 
     def package(self):
-        self.copy(pattern="COPYING", src=self._source_subfolder, dst="licenses")
-        self.copy(pattern="make", src=self._source_subfolder, dst="bin", keep_path=False)
-        self.copy(pattern="*gnumake.exe", src=self._source_subfolder, dst="bin", keep_path=False)
+        copy(self, "COPYING", src=self.source_folder, dst=os.path.join(self.package_folder, "licenses"))
+        for make_exe in ("make", "*gnumake.exe"):
+            copy(self, make_exe, src=self.source_folder, dst=os.path.join(self.package_folder, "bin"), keep_path=False)
 
     def package_info(self):
+        self.cpp_info.includedirs = []
         self.cpp_info.libdirs = []
 
         make = os.path.join(self.package_folder, "bin", "gnumake.exe" if self.settings.os == "Windows" else "make")
+        self.conf_info.define("tools.gnu:make_program", make)
```

Comment:
I thought you wanted these removed like with ninja? #13853 why is this different?

Reply:
it's the path of make itself here, it doesn't manipulate settings of other tools

Reply:
Can we please pick one please?

- using tool recipes from CCI will update your conf setting
- Or it won't

We'll write the docs and that will be the rule moving forward since some do some don't its way too confusing.

Reply:
It's not confusing. If there is a conan conf related to path of a specific tool, recipe packaging this tool will obviously set this conf, that's all.

Reply:
Hmmm that sounds easier with more examples. 

So no setting conf that change behavior of the client - that reserved for the end consumer explicitly opting in